packages/ui/src/Table.tsx

```diff
@@ -0,0 +1,384 @@
+
+/**
+ * Table props.
+ */
+export interface TableProps extends React.ComponentPropsWithoutRef<'table'> {
+  /**
+   * Required data prop, should be array of objects.
+   */
+  data: TableData[]
+  /**
+   * Required columns prop, used to layout the content in cells and columns
+   */
+  columns: Record<string, ColumnData>
+  /**
+   * Renders Loader instead of table content.
+   */
+  isLoading?: boolean
+  /**
+   * Optional message when there is no data present
+   */
+  noDataMessage?: string
+  /**
+   * Optional title that will appear above header columns
+   */
+  titleComponent?: string | React.ReactNode
+  /**
+   * Optional footer that will appear below table body
+   */
+  footer?: React.ReactNode
+  /**
+   * Optional Footer className to override any default styling
+   */
+  footerClassName?: string
+}
+
+/**
+ * Row props.
+ */
+interface RowProps
+  extends Pick<TableProps, 'columns'>,
+    React.ComponentPropsWithoutRef<'tr'> {
+  /**
+   * Row index.
+   */
+  rowIndex: number
+  /**
+   * Renders Loader instead of table content.
+   */
+  rowData: TableData
+  /**
+   * Boolean if table title is present
+   */
+  withTitle?: boolean
+  /**
+   * Boolean if table title is a component or a string
+   */
+  isTitleAString?: boolean
+}
+
+const Row = ({
+  columns,
+  rowData,
+  rowIndex,
+  withTitle,
+  isTitleAString,
+  ...props
+}: RowProps) => {
+  const [isRowExpanded, setIsRowExpanded] = useState(false)
+  const [expandedRowData, setExpandedRowData] = useState<TableData>({})
+
+  const onRowExpand = (data: TableData) => {
+    setExpandedRowData({ ...data })
+
+    if (data) {
+      setIsRowExpanded(true)
+    } else {
+      setIsRowExpanded(false)
+    }
+  }
+
+  const Row = (
+    <tr {...props}>
+      {Object.values(columns).map(({ Cell, alignText = 'center' }, index) => {
+        if (!Cell) {
+          return null
+        }
+
+        return (
+          <Column
+            key={index}
+            alignText={alignText}
+            withTitle={withTitle}
+            isTitleAString={isTitleAString}
+          >
+            <Cell onRowExpand={onRowExpand} data={rowData} index={rowIndex} />
+          </Column>
+        )
+      })}
+    </tr>
+  )
+
+  return (
+    <>
+      {Row}
+      {isRowExpanded && (
+        <tr>
+          {Object.values(columns).map(({ ExpandedCell }, index) => {
+            if (!ExpandedCell) {
+              return null
+            }
+
+            return (
+              <ExpandedColumn
+                colSpan={Object.values(columns).length}
+                key={index}
+              >
+                <ExpandedCell
+                  data={{ ...expandedRowData, ...rowData }}
+                  index={rowIndex}
+                />
+              </ExpandedColumn>
+            )
+          })}
+        </tr>
+      )}
+    </>
+  )
+}
+
+export const Table = ({
+  data,
+  columns,
+  isLoading,
+  noDataMessage = '',
+  titleComponent = '',
+  footer,
+  footerClassName = '',
```

Comment:
Well, that class was meant to be allowed to edit `tfoot` and `td` inside the footer while wrapping the component that will be passed into. If you want, I can remove it and leave it as it is.  We also said that we need some className to alter the footer if needed: https://bloxico.atlassian.net/browse/BLOX-165 (footer section).

Reply:
oh, yes, sry on confusion, I said className for footer, but what i meant is hardcoded className, because I thought, maybe you will add footer as a div, not some specific element, we can remove this prop fully. I'll do that part for this task.